### PR TITLE
[quant] Fix dimension for output of batchnorm 1d

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59264 [quant] Fix dimension for output of batchnorm 1d**

Summary:
fixes: https://github.com/pytorch/pytorch/issues/59200
Previously batchnorm 1d did unsqueeze twice but only squeeze once before return when the dimension
for input Tensor is 2, this PR adds an extra squeeze

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28810597](https://our.internmc.facebook.com/intern/diff/D28810597)

### CODE REVIEW

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28810597).

**Comment by pmeier** [comment]
@jerryzh168 You probably disable the 1d case in 

https://github.com/pytorch/pytorch/blob/6d45d7a6c331ddb856ac34a76bcd3613aa05185b/test/quantization/core/test_quantized_op.py#L2104

and

https://github.com/pytorch/pytorch/blob/6d45d7a6c331ddb856ac34a76bcd3613aa05185b/test/quantization/core/test_quantized_op.py#L2155

They only pass now because there is a bug in the underlying comparison mechanism. Try put a `assert qx.shape == qy.shape` in there and see for yourself. After #59067 has landed, the test will (correctly) fail if not fixed.

**Comment by jerryzh168** [comment]
> @jerryzh168 You probably need to disable the 1d case in
> 
> https://github.com/pytorch/pytorch/blob/6d45d7a6c331ddb856ac34a76bcd3613aa05185b/test/quantization/core/test_quantized_op.py#L2104
> 
> and
> 
> https://github.com/pytorch/pytorch/blob/6d45d7a6c331ddb856ac34a76bcd3613aa05185b/test/quantization/core/test_quantized_op.py#L2155
> 
> They only pass now because there is a bug in the underlying comparison mechanism. Try put a `assert qx.shape == qy.shape` in there and see for yourself. After #59067 has landed, the tests will (correctly) fail if not fixed.

 makes sense, this PR would fix the issue though, if we land this PR before yours it would not fail I think?

**Comment by pmeier** [comment]
If I'm not mistaken you only need to remove the `2` from

https://github.com/pytorch/pytorch/blob/6d45d7a6c331ddb856ac34a76bcd3613aa05185b/test/quantization/core/test_quantized_op.py#L2106

as well as the `==2` check here

https://github.com/pytorch/pytorch/blob/6d45d7a6c331ddb856ac34a76bcd3613aa05185b/test/quantization/core/test_quantized_op.py#L2130

**Comment by jerryzh168** [comment]
> If I'm not mistaken you only need to remove the `2` from
> 
> https://github.com/pytorch/pytorch/blob/6d45d7a6c331ddb856ac34a76bcd3613aa05185b/test/quantization/core/test_quantized_op.py#L2106
> 
> as well as the `==2` check here
> 
> https://github.com/pytorch/pytorch/blob/6d45d7a6c331ddb856ac34a76bcd3613aa05185b/test/quantization/core/test_quantized_op.py#L2130

are you saying I don't need to add an extra test for batchnorm 1d? I can remove that

**Comment by pmeier** [comment]
Oh sorry, I misunderstood the intent of the PR. Ignore my earlier comment. Before you land this PR though, could you locally put in `assert qx.shape == qy.shape` to make sure this is actually passing?

**Comment by jerryzh168** [comment]
> Oh sorry, I misunderstood the intent of the PR. Ignore my earlier comment. Before you land this PR though, could you locally put in `assert qx.shape == qy.shape` to make sure this is actually passing?

yes, will do that

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28810597).

**Review by pmeier — APPROVED** [review]
LGTM! If it passes locally with the suggested check, it should be fine.

**Comment by jerryzh168** [comment]
> LGTM! If it passes locally with the suggested check, it should be fine.

yes it did pass the shape check locally

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28810597).
